org.eclipse.xtext/src/org/eclipse/xtext/diagnostics/Diagnostic.java

```diff
@@ -54,4 +54,18 @@
 	 */
 	int getLength();
 
+    /**
+     * Returns the end line location of the issue within the source.
+     * Line <code>1</code> is the first line.
+     * @return the end line location of the issue.
+     */
+    int getLineEnd();
```

Comment:
ihmo this is a breaking api change

Reply:
Yes, but this class and class `Issue` have the annotation `@noimplement`. Maybe that would be okay then? However, I add `@since 2.21` to the new methods.

Reply:
i see. @szarnekow what do you think?

Reply:
In general I'm fine with these being added here. I'm still a little but undecided if I'd like these to have default impls.